client/src/lib/data.ts

```diff
@@ -28,6 +28,64 @@ export function readToken(): string | undefined {
   return (JSON.parse(auth) as Auth).token;
 }
 
+type SignUpParams = {
+  username: string;
+  password: string;
+  displayName: string | undefined;
+};
+export async function signUp(params: SignUpParams): Promise<User> {
+  const { username, password, displayName } = params;
+  const req = {
+    method: 'post',
+    body: JSON.stringify({
+      username,
+      password,
+      displayName: displayName ? displayName : username,
```

Comment:
This could be shortened to 
`displayName || username`
or
`displayName ?? username`